### PR TITLE
Allow recursive nesting of AdditionalProperties in Attributes

### PR DESCRIPTION
**Pull request by rien-vroom-cquential**
Since nested AdditionalProperties have been allowed for a while through #706 also allow it through Attributes and not only with annotations.

### CODE REVIEW

File: src/Attributes/AdditionalProperties.php [hunk 1]
```diff
@@ -98,6 +99,7 @@ public function __construct(
             'allOf' => $allOf ?? Generator::UNDEFINED,
             'anyOf' => $anyOf ?? Generator::UNDEFINED,
             'oneOf' => $oneOf ?? Generator::UNDEFINED,
+            'additionalProperties' => $additionalProperties ?? Generator::UNDEFINED,
```
Comment (DerManoMann) [inline src/Attributes/AdditionalProperties.php:102]: Have you tested this? All other nested objects are passed into the parent in this line:
```
            'value' => $this->combine($items, $discriminator, $externalDocs, $attachables),
```

**Review by DerManoMann — CHANGES_REQUESTED** [review]
It would be great if you could add a simple example of the nesting in the `tests/Fixtures/Scratch` folder.

Something like this (still missing the nesting):
```php
// NestedAdditionalProperties.php
<?php declare(strict_types=1);

/**
 * @license Apache 2.0
 */

namespace OpenApi\Tests\Fixtures\Scratch;

use OpenApi\Attributes as OAT;

#[OAT\Schema(
    additionalProperties: new OAT\AdditionalProperties(
        type: 'string',
        schema: new OAT\Schema(
            type: 'object',
        )
    )
)]
class NestedAdditionalPropertiesModel
{
}

#[OAT\Info(
    title: 'Nested Additional Properties',
    version: '1.0'
)]
#[OAT\Get(
    path: '/api/endpoint',
    description: 'An endpoint',
    responses: [new OAT\Response(response: 200, description: 'OK', ref: NestedAdditionalPropertiesEndpoint::class)]
)]
class NestedAdditionalPropertiesEndpoint
{
}
```

Running the scratch test will create the corresponding `.yaml` file which you should check for correctness.

File: src/Attributes/AdditionalProperties.php (hunk 1 above)
Comment (rien-vroom-cquential) [reply src/Attributes/AdditionalProperties.php:102]: For the value of additionalProperties i just copied the "OAT\Schema" style. 
I have changed it to an additional argument of the `combine()` method and the endresult is the same.
Aka it generates the same YAML file.

Change is on its way

**Review by DerManoMann — APPROVED** [review]
Just needs a code style fix then should be good to go.

Thanks.

**Comment by DerManoMann** [comment]
Thanks @rien-vroom-cquential

**Pull request merged**